### ISSUE TITLE
Redirect to secondary login page

### ISSUE THREAD

**Issue by goetzk** [tendenci/tendenci#656]
Hi,
I'm after your input/suggestions as my upstream for Tendenci; specifically my login function.

I'm deploying Tendenci with django-otp and I want to change the login flow to include a mandatory second view.

Login page -> custom view -> value of ?next= or I suppose LOGIN_REDIRECT_URL.

At the moment I'm thinking about deploying this change to Tendenci:

```
diff --git a/tendenci/apps/accounts/views.py b/tendenci/apps/accounts/views.py
index ef1f013..ed04972 100644
--- a/tendenci/apps/accounts/views.py
+++ b/tendenci/apps/accounts/views.py
@@ -18,9 +18,22 @@ from tendenci.apps.accounts.forms import PasswordResetForm
 
 
 @ssl_required
-def login(request, form_class=LoginForm, template_name="account/login.html"):
+def login(request, form_class=LoginForm, template_name="account/login.html", second_view_name=None):
+    """Manage logins to Tendenci.
 
-    redirect_to = request.GET.get('next', u'')
+    Takes request plus three user configurable kwargs:
+     - form_class: A login form to use; must be a forms.Form instance
+     - template_name: Name of login template html file including path
+     - second_view_name: A view to redirect to for multi page logins
+    """
+    if second_view_name:
+        # If a second view will be used for authentication we can't take them
+        # directly where they want to go so we will try and store it for later.
+        # If they didn't specify a preference, we'll take them to home
+        request.session['final_destination'] = request.GET.get('next', u'/')
+        redirect_to = reverse(second_view_name)
+    else:
+        redirect_to = request.GET.get('next', u'')
 
     if request.method == "POST":
         default_redirect_to = getattr(settings, "LOGIN_REDIRECT_URLNAME", None)
```

This precludes me from using LOGIN_REDIRECT_URL but is sufficient for my needs.

I'm wondering if:
- There is some way to use the Tendenci login view that I'm not seeing, which would allow me to skip this customisation
- If you would be willing to take a PR which does the above; changed or unchanged.

Thanks,
Karl.

**Comment by jennyq** [comment]
Hi @goetzk, 
I think it's fine to make a PR like that. Can you share a bit more regarding your implementation with `django-otp`? Is the `second_view_name` going to be used for two-factor authentication?  Have you looked at `django-two-factor-auth`? It might be a better option. Certainly, it won't work with the current version of tendenci yet.

**Comment by goetzk** [comment]
Hi @jennyq 

I spent more time looking at this today and came up with a better solution (at least, I think so). Its a bit of middleware to do the enforcing of users for me, I emailed the django-otp group about it here.
https://groups.google.com/forum/#!topic/django-otp/4kUj_pUAas0

In answer to your question, I have given two-factor a cursory look, but won't be looking at it in detail for a little while yet and it will need customisation for my use case (It doesn't support pluggable authentication backends).

thanks for having a look at this request, I will let you know if I have a chance to look at django-two-factor-auth.
kk
